packages/@aws-cdk/aws-rds/test/serverless-cluster.test.ts

```diff
@@ -863,6 +863,221 @@ describe('serverless cluster', () => {
+        snapshotIdentifier: 'my-snapshot',
+      });
+
+      // THEN
+      expect(stack).toHaveResource('AWS::RDS::DBCluster', {
+        Properties: {
+          Engine: 'aurora-mysql',
+          DBClusterParameterGroupName: 'default.aurora-mysql5.7',
+          DBSubnetGroupName: {
+            Ref: 'ServerlessDatabaseSubnets5643CD76',
+          },
+          EngineMode: 'serverless',
+          SnapshotIdentifier: 'my-snapshot',
+          StorageEncrypted: true,
+          VpcSecurityGroupIds: [
+            {
+              'Fn::GetAtt': [
+                'ServerlessDatabaseSecurityGroupB00D8C0F',
+                'GroupId',
+              ],
+            },
+          ],
+        },
+        DeletionPolicy: 'Snapshot',
+        UpdateReplacePolicy: 'Snapshot',
+      }, ResourcePart.CompleteDefinition);
+
+    });
+
+    test('can generate a new snapshot password', () => {
+      const stack = testStack();
+      const vpc = new ec2.Vpc(stack, 'VPC');
+
+      // WHEN
+      new ServerlessClusterFromSnapshot(stack, 'ServerlessDatabase', {
+        engine: DatabaseClusterEngine.AURORA_MYSQL,
+        vpc,
+        snapshotIdentifier: 'mySnapshot',
+        credentials: SnapshotCredentials.fromGeneratedSecret('admin', {
+          excludeCharacters: '"@/\\',
+        }),
+      });
+
+      // THEN
+      expect(stack).toHaveResourceLike('AWS::RDS::DBCluster', {
+        MasterUsername: ABSENT,
+        MasterUserPassword: {
+          'Fn::Join': ['', [
+            '{{resolve:secretsmanager:',
+            { Ref: 'ServerlessDatabaseSecret813910E98ee0a797cad8a68dbeb85f8698cdb5bb' },
+            ':SecretString:password::}}',
+          ]],
+        },
+      });
+      expect(stack).toHaveResource('AWS::SecretsManager::Secret', {
+        Description: {
+          'Fn::Join': ['', ['Generated by the CDK for stack: ', { Ref: 'AWS::StackName' }]],
+        },
+        GenerateSecretString: {
+          ExcludeCharacters: '\"@/\\',
+          GenerateStringKey: 'password',
+          PasswordLength: 30,
+          SecretStringTemplate: '{"username":"admin"}',
+        },
+      });
+
+    });
+
+    test('fromGeneratedSecret', () => {
+      const stack = testStack();
+      const vpc = new ec2.Vpc(stack, 'VPC');
+
+      // WHEN
+      new ServerlessClusterFromSnapshot(stack, 'ServerlessDatabase', {
+        engine: DatabaseClusterEngine.AURORA_MYSQL,
+        vpc,
+        snapshotIdentifier: 'mySnapshot',
+        credentials: SnapshotCredentials.fromGeneratedSecret('admin', {
+          excludeCharacters: '"@/\\',
+        }),
+      });
+
+      // THEN
+      expect(stack).toHaveResourceLike('AWS::RDS::DBCluster', {
+        MasterUsername: ABSENT,
+        MasterUserPassword: {
+          // logical id of secret has a hash
+          'Fn::Join': ['', ['{{resolve:secretsmanager:', { Ref: 'ServerlessDatabaseSecret813910E98ee0a797cad8a68dbeb85f8698cdb5bb' }, ':SecretString:password::}}']],
+        },
+      });
+
+    });
```

Comment:
Sorry, what's the difference between this test, and the one above it?

Reply:
You're right, this test is a duplicate. I will remove it.
I copied the tests from `instance.test.ts` adjusted it to `ServerlessCluster`. From my point of view, it is a duplicate [there](https://github.com/aws/aws-cdk/blob/890c4c5f9d4835b39c8448e47c781b5e8334cd60/packages/%40aws-cdk/aws-rds/test/instance.test.ts#L338-L355) as well. Should I remove it in this PR too?

Reply:
Nah, that's fine. I don't want to make this PR bigger. Appreciate the thoughtfulness though!